default/python/AI/ShipDesignAI.py

```diff
@@ -854,6 +861,15 @@ def update_stats(self, ignore_species=False):
         self.production_cost = local_cost_cache.get(self.hull.name, self.hull.productionCost(fo.empireID(), self.pid))
         self.production_time = local_time_cache.get(self.hull.name, self.hull.productionTime(fo.empireID(), self.pid))
 
+        # calculate hull stat effects
+        if self.species and not ignore_species:
+            self.design_stats.fuel += _get_species_fuel_bonus(self.species)
+        # TODO tech hull fuel effects
+        if self.design_stats.fuel < 0:
+            self.design_stats.fuel = 0
+        else:
+            self.design_stats.fuel *= self._hull_efficiency()
```

Comment:
Seems more maintainable / in alignment with FOCS script to multiply the hull efficiency at a single point at the end of this function rather than applying it to the hull and to the parts separately unless I missed something.

Reply:
You missed that I intended another effect implementation which would not have applied to the hull base fuel capacity. But that implementation was not really feasable and i am happy with how the current one turned out.

So, yes, so be it